### PR TITLE
Don't use rexml for the JUnit formatter

### PR DESCRIPTION
**Pull request by Earlopain**
Closes https://github.com/Shopify/erb_lint/issues/371
Follows https://github.com/rubocop/rubocop/pull/13165

This leaves the output mostly the same, except that text is no longer wrapped in CDATA. Some things possibly need to be escaped now, and CDATA has different escaping needs. So, just use the same escaping everywhere

### CODE REVIEW

File: spec/erb_lint/cli_spec.rb [hunk 1]
```diff
@@ -4,7 +4,6 @@
 require "spec_utils"
 require "erb_lint/cli"
 require "erb_lint/cache"
-require "pp"
```
Comment (Earlopain) [inline spec/erb_lint/cli_spec.rb:7]: This is needed. Somehow the load order is now different, so this now runs into https://github.com/fakefs/fakefs/issues/215

File: lib/erb_lint/reporters/junit_reporter.rb [hunk 2]
```diff
@@ -1,111 +1,61 @@
...
-        failures = stats.found
-        testsuite_element = REXML::Element.new("testsuite", nil, CONTEXT)
-        testsuite_element.add_attribute("name", "erblint")
-        testsuite_element.add_attribute("tests", tests.to_s)
-        testsuite_element.add_attribute("failures", failures.to_s)
-
-        testsuite_element.add_element(create_properties)
+        puts %(  <properties>)
+        PROPERTIES.each do |key, value|
+          puts %(    <property name="#{key}" value="#{value}"/>)
```
Comment (etiennebarrie) [inline lib/erb_lint/reporters/junit_reporter.rb:30]: There's little risk but we should still be escaping all the attributes we pass IMO.

**Review by etiennebarrie — COMMENTED** [review]
Thanks, a few comments about escaping.
What you're doing seems safe but still, let's always escape whole strings.

**Comment by Earlopain** [comment]
>  What you're doing seems safe but still, let's always escape whole strings.

Yeah, sure. It shouldn't matter much. I just added escaping to pretty much everything now.

**Pull request merged**